### PR TITLE
Score storage

### PR DESCRIPTION
**Pull request by alvestrand**
This PR adds the function of maintaining a common Git repository for encode results from all users.

### CODE REVIEW

File: bin/export_improved_scores [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+#
+import argparse
+import sys
+
+import encoder
+import optimizer
+import pick_codec
+import score_tools
+
+def ImprovementsInOneParameterSet(my_optimizer, filename):
```
Comment (phoglund) [inline bin/export_improved_scores]: Nit: 2 blank lines

File: bin/export_improved_scores [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  results = []
+  for encoding in cache.AllScoredEncodingsForEncoder(my_encoder):
+    new_score = my_optimizer.Score(encoding)
+    try:
+      old_score = my_optimizer.Score(encoding, scoredir='score_storage')
+    except (encoder.Error, KeyError):
+      # encoder.Error can happen if old score is broken.
+      # KeyError can happen if old score is missing some now-mandatory
+      # field.
+      # In both cases, the new score should be used.
```
Comment (phoglund) [inline bin/export_improved_scores]: Nit: pull up line 38 to line 37.

File: bin/export_improved_scores [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      if new_score > 0.0:
+        results.append(encoding)
+      continue
+    if new_score > old_score:
```
Comment (phoglund) [inline bin/export_improved_scores]: Suggestion: move this into the try block and remove the continue, should do the same thing right?

File: bin/export_improved_scores [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      results.append(encoding)
+  return results
+
+def FindImprovedScores(codecs, criterion):
```
Comment (phoglund) [inline bin/export_improved_scores]: Nit: two blank lines

File: bin/export_improved_scores [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    codec = pick_codec.PickCodec(codec_name)
+    my_optimizer = optimizer.Optimizer(codec,
+        score_function=score_tools.PickScorer(criterion))
+    filenames = my_optimizer.context.cache.AllEncoderFilenames()
+    for filename in filenames:
+      improvement_list.extend(ImprovementsInOneParameterSet(my_optimizer,
+                                                            filename))
+  return improvement_list
+
+def main():
```
Comment (phoglund) [inline bin/export_improved_scores]: Nit: two blank lines

File: bin/import_scores [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Checks out the scores stored in the central repo.
+#
+set -e
+
+if [ ! $WORKDIR ]; then
+  echo "init.sh not run"
```
Comment (phoglund) [inline bin/import_scores]: "Please run init.sh first."

**Comment by phoglund** [comment]
lgtm % nits

File: bin/export_improved_scores (hunk 3 above)
Comment (alvestrand) [inline bin/export_improved_scores]: Good idea.

File: bin/export_improved_scores (hunk 2 above)
Comment (alvestrand) [inline bin/export_improved_scores]: Simplified text.

File: bin/import_scores (hunk 6 above)
Comment (alvestrand) [inline bin/import_scores]: Done.

**Pull request merged**